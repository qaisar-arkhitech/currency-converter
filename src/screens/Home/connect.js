import {connect} from "react-redux"

import {selectStyleableTheme} from "../../redux/reducers/themes/selectors"

import {createLoadAction} from "../../redux/reducers/conversion/actionCreators"

const mapStateToProps = (state) => ({
  styleableTheme: selectStyleableTheme(state),
})

const mapDispatchToProps = {
  getRates: createLoadAction,
}

export default connect(mapStateToProps, mapDispatchToProps)
