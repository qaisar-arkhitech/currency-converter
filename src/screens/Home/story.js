import React from "react"
import Home from "."

export default {
  title: "Home",
  storyFn() {
    return (
      <div>
        <Home />
      </div>
    )
  },
}
