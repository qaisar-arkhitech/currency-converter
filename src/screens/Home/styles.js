import {StyleSheet} from "react-native"

// Home: Styling here
const styles = StyleSheet.create({
  root: {
    flex: 1,
  },
  header: {
    alignItems: "flex-end",
    marginHorizontal: 20,
  },
  input: {
    padding: 10,
    fontSize: 16,
    backgroundColor: "#000",
    height: 40,
  },
})

export default styles
