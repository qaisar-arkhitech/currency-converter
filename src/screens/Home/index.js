import React, {useState} from "react"
import flow from "lodash/flow"
import {
  View,
  StatusBar,
  ScrollView,
  TouchableOpacity,
  TextInput,
} from "react-native"
import Entypo from "react-native-vector-icons/Entypo"
import PropTypes from "prop-types"
import {SafeAreaView} from "react-native-safe-area-context"
import themeStyles from "./styles"
import connect from "./connect"
import KeyboardSpacer from "../../components/KeyboardSpacer"

// Home Component content
export const Home = ({navigation, styleableTheme}) => {
  const [scrollEnabled, setScrollEnabled] = useState(false)
  const styles = themeStyles(styleableTheme)
  return (
    <View style={styles.root}>
      <StatusBar barStyle="default" backgroundColor={styleableTheme[500]} />
      <ScrollView scrollEnabled={scrollEnabled}>
        <SafeAreaView style={styles.header}>
          <TouchableOpacity onPress={() => navigation.push("Options")}>
            <Entypo name="cog" size={32} color="#fff" />
          </TouchableOpacity>
        </SafeAreaView>
        <View>
          <TextInput style={styles.input} />
        </View>
        <KeyboardSpacer onToggle={(visible) => setScrollEnabled(visible)} />
      </ScrollView>
    </View>
  )
}

// Home Proptypes
Home.propTypes = {
  navigation: PropTypes.object.isRequired,
  styleableTheme: PropTypes.object,
}

// Home Default props
Home.defaultProps = {
  styleableTheme: {},
}

export default flow([connect])(Home)
