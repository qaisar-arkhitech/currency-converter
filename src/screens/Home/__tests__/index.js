import React from "react"
import renderer from "react-test-renderer"
import Home from ".."

const createProps = () => ({
  classes: {},
})

describe("<Home />", () => {
  it("Does not explode", () => {
    const props = createProps()
    const component = renderer.create(<Home {...props} />).toJSON()
    expect(component).toMatchSnapshot()
  })
})
