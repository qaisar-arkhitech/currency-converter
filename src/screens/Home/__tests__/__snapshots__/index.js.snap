// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Home /> Does not explode 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#e6f7ee",
      "flex": 1,
    }
  }
>
  <RCTScrollView
    scrollEnabled={false}
  >
    <View>
      <RNCSafeAreaView
        style={
          Object {
            "alignItems": "flex-end",
            "marginHorizontal": 20,
          }
        }
      >
        <View
          accessible={true}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "opacity": 1,
            }
          }
        >
          <Text
            allowFontScaling={false}
            style={
              Array [
                Object {
                  "color": "#fff",
                  "fontSize": 32,
                },
                undefined,
                Object {
                  "fontFamily": "Entypo",
                  "fontStyle": "normal",
                  "fontWeight": "normal",
                },
                Object {},
              ]
            }
          >
            ï…œ
          </Text>
        </View>
      </RNCSafeAreaView>
      <View>
        <TextInput
          allowFontScaling={true}
          rejectResponderTermination={true}
          style={
            Object {
              "backgroundColor": "#000",
              "fontSize": 16,
              "height": 40,
              "padding": 10,
            }
          }
          underlineColorAndroid="transparent"
        />
      </View>
      <View
        style={
          Array [
            Object {
              "bottom": 0,
              "left": 0,
              "right": 0,
            },
            Object {
              "height": 0,
            },
            Object {},
          ]
        }
      />
    </View>
  </RCTScrollView>
</View>
`;
